app/controllers/concerns/foreman/controller/bruteforce_protection.rb

```diff
@@ -1,20 +1,19 @@
 module Foreman::Controller::BruteforceProtection
   extend ActiveSupport::Concern
 
-  def count_login_failure
-    Rails.cache.write("failed_login_#{request.ip}", get_login_failures + 1, :expires_in => 5.minutes)
-  end
-
-  def get_login_failures
-    Rails.cache.fetch("failed_login_#{request.ip}") {0} if request.ip.present?
+  included do
+    delegate :count_login_failure, :get_login_failures, :bruteforce_attempt?, :log_bruteforce, to: :bruteforce_protection
```

Comment:
Note to self: this has bruteforce_attempt? which is overridden below and should be removed.